### DISCUSSION TITLE
outline view extremely slow

### DISCUSSION THREAD

**Discussion by amigo421** [microsoft/vscode-cpptools#7154] [Q&A]
any idea how to speed up outline view ?
if I work with some cpp file it is loading symbols too long ~30 mins (not seconds! but minutes!) current file which I look at for now is ~500 lines, at the same time I can navigate there (F12) 

there is also another issue: "Initializing tag parsing" - if I see such message then looks like language engine hangs, if I will wait even a day nothing happens, and I need to call reset intellisense engine , I really need this daily ((

vscode and extensions are latest from stable channels

Ctrl+T search for symbol in a project , I can wait for many hours

**Comment by sean-mcmanus** [comment]
We've reimplemented the outline view for our pending 1.3.0-insiders release, so your issue may be fixed with that.

However, it sounds like you may be hitting a problem with tag parsing or the database. If you see "initializing tag parser" for too long then that means the database can't be access, maybe due to another cpptools process having a lock on it.

However, if you're seeing an issue where the outline view is not updating as it should, that could be the same issue I mentioned at https://github.com/microsoft/vscode-cpptools/issues/7118 ...we're not sure of a good way to repro it yet.
